backend/tournesol/views/video.py

```diff
@@ -41,8 +41,12 @@ def list(self, request, *args, **kwargs):
         queryset = Video.objects.all()
         search = request.query_params.get('search')
         if search:
-            queryset = queryset.filter(Q(name__icontains=search) |
-                                       Q(description__icontains=search))
+            #  search keywork is about the same as icontains but is more efficient with a psql db
+            queryset = queryset.filter(
+                Q(name__search=search) |
+                Q(description__search=search) |
+                Q(tags__name__search=search)
+            )
```

Comment:
In what sense is it more efficient exactly?  
As far as I can see, it is indeed more robust to different query terms, as a tokenizer is used by Postgres to compare the data with the query.  But it also seems to be significantly slower than `__icontains`.

According to [Django documentation](https://docs.djangoproject.com/en/3.2/ref/contrib/postgres/search/#performance):
> Special database configuration isn’t necessary to use any of these functions, however, if you’re searching more than a few hundred records, you’re likely to run into performance problems. [...]

To make this approach work on production data, we would probably need to index the search vectors, for example using a [`SearchVectorField`](https://docs.djangoproject.com/en/3.2/ref/contrib/postgres/search/#searchvectorfield).